lib/spack/spack/cmd/find.py

```diff
@@ -96,6 +97,14 @@ def setup_parser(subparser):
                            action='store_true',
                            help='show fully qualified package names')
 
+    subparser.add_argument(
+        '--start-date',
+        help='earliest date of installation [YYYY-MM-DD HH:MM:SS]'
```

Comment:
I think the format here is too cumbersome.  Most people don't care about the seconds.  It should really let you specify something telescoping like `YYYY`, `YYYY-MM`, `YYYY-MM-DD`, etc.  Or at least just `YYYY-MM-DD`.  If you want to get fancy it would be really cool to invert `llnl.util.lang.pretty_date` and allow, e.g., `--since yesterday` or `--since "a month ago"`, but I don't think that's necessary.

Reply:
I implemented a minimal parser for this one. If in the future we want a better solution, we might consider [this](https://github.com/scrapinghub/dateparser) (thanks to @nazavode for the pointer).